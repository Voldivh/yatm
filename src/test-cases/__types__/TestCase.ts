import type Requirement from '../../requirements/__types__/Requirement'

type TestCase = {
  dimensions: {
    [dimension: string]: string
  }
  generation: number
} & Requirement

export default TestCase
