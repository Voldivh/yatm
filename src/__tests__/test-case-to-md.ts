import testCaseToMd from '../test-case-renderer/test-case-to-md'

import TestCase from '../__types__/TestsCase'

describe('testCaseToMd', () => {
  it('should render a test case', () => {
    const testCase: TestCase = {
      name: 'test case',
      platform: 'jammy',
      dds: 'fastdds',
      installType: 'source',
      checks: [
        {
          name: 'check',
          try: [
            {
              note: 'try',
              terminal: 1,
              stderr: 'stderr',
            },
          ],
          expect: [
            {
              note: 'expect',
              terminal: 1,
              stdout: 'stdout',
            },
            {
              note: 'expect',
              imageUrl: 'image url',
            },
          ],
        },
        {
          name: 'Name only check',
        },
        {
          name: 'Name only check 2',
        },
      ],
    }

    expect(testCaseToMd(testCase)).toMatchSnapshot()
  })
})
