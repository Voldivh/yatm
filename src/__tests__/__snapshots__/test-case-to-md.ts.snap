// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testCaseToMd should render a test case 1`] = `
"# test case

## Setup
- Platform: jammy
- Install Type: source
- DDS: fastdds

## Checks
- [ ] check

  <details><summary>details</summary>
  
    **Try**
  
    1.
       try
       \`\`\`bash
       # stderr in terminal 1
       stderr
       \`\`\`
  
    **Expect**
  
    1.
       expect
       \`\`\`bash
       # stdout in terminal 1
       stdout
       \`\`\`
    1.
       expect
       ![](image url)
       
  
  </details>
- [ ] Name only check
- [ ] Name only check 2"
`;
